src/types/estree.d.ts

```diff
@@ -0,0 +1,7 @@
+import { JSONNode } from "jsonc-eslint-parser";
+
+declare module "estree" {
+	interface NodeMap {
+		JSONNode: JSONNode;
+	}
+}
```

Comment:
[Praise] _Oh_ I did not at all think of doing this. Very nice, what a clean solution!

Maybe it makes sense to propose adding this to https://github.com/ota-meshi/jsonc-eslint-parser's docs?